compiler/rustc_symbol_mangling/src/v0.rs

```diff
@@ -70,6 +72,54 @@ pub(super) fn mangle<'tcx>(
     std::mem::take(&mut cx.out)
 }
 
+pub fn mangle_internal_symbol<'tcx>(tcx: TyCtxt<'tcx>, item_name: &str) -> String {
+    if item_name == "rust_eh_personality" {
+        // rust_eh_personality must not be renamed as LLVM hard-codes the name
+        return "rust_eh_personality".to_owned();
+    } else if item_name == "__rust_no_alloc_shim_is_unstable" {
+        // Temporary back compat hack to give people the chance to migrate to
+        // include #[rustc_std_internal_symbol].
+        return "__rust_no_alloc_shim_is_unstable".to_owned();
```

Comment:
Question: should we have a cleanup issue about planning to eventually deprecate this? Unless we already have one.

Reply:
The tracking issue for this symbol is https://github.com/rust-lang/rust/issues/123015